carbonmark-api/src/utils/helpers/carbonProjects.utils.ts

```diff
@@ -12,12 +12,98 @@ type Args = {
   registryProjectId: string; //e.g 1121
 };
 
+const standards = ["VCS", "PURO", "ICR"] as const;
+
+/** Known supported standards on-chain */
+export type Standard = (typeof standards)[number];
+/** Project id number as it appears in the registry, stringified to preserve leading zeros @example "191" */
+export type RegistryProjectId = string;
+/** Credit vintage @example 2008 */
+export type Vintage = string;
+/** @example "VCS-191" */
+export type ProjectIdentifier = `${Standard}-${RegistryProjectId}`;
+/** Full credit id, converted to uppercase @example "VCS-191-2008" */
+export type CreditIdentifier = `${ProjectIdentifier}-${Vintage}`;
+
+type UntypedParams = {
+  standard: string;
+  vintage: string;
+  registryProjectId: string;
+};
+
+/* eslint-disable @typescript-eslint/consistent-type-assertions -- type guards */
```

Comment:
😬 ok for the typeguards but we shouldn't need to cast. i.e `as Standard`

Reply:
No way around it that I can tell, if I want the constructor to accept `string` (i.e. query params or cms data) but I want the final class properties to have the interpolated string types for example `${VCS|PURO|ICR}-${number}`